### ISSUE TITLE
Do not expose an open query interface by default

### ISSUE THREAD

**Issue by jsumners** [apollographql/apollo-server#4809]
Relying on `NODE_ENV !== 'production'` as a mechanism to enable an open query interface is not an acceptable method of enabling such an easily misused feature. A proper safe default is to _always_ have this feature disabled unless it has been explicitly enabled by the developer.

**Comment by glasser** [comment]
I don't know if you mean "displaying a query UI" or "exposing the schema over introspection" here, but we're pretty happy with the default behavior of Apollo Server and the various ways of customizing these behaviors.

**Comment by jsumners** [comment]
You're exposing a developer testing tool by default under the assumption that code is running in a development environment unless a magic environment variable is set. That's the wrong way to do it. Code should be assumed to be running in production first. Developer centric features should be opted in specifically by the developer.

**Comment by glasser** [comment]
I hear that. This has been the approach taken by this software for about four years, and the tradeoffs have seemed appropriate so far.  We're certainly open to reconsidering the defaults in a future major version if we get a lot of feedback, but frankly we got the opposite feedback when we shifted the default from "display GraphQL Playground in development mode" to "display a splash screen in development mode" between AS2 and AS3.
